### DISCUSSION TITLE
POST own endpoint through load balancer

### DISCUSSION THREAD

**Discussion by NixBiks** [fastapi/fastapi#5957] [Questions]
I'm wondering if it's possible to easily make a POST one of the fastapi applications own endpoints? Why would I do this? To run several downstream tasks in parallel (let the load balancer take care of it). 

Currently I'm code generating a python client based on the openapi spec and using that. It's a bit cumbersome (two way dependency between app and codegen) and there might be better ways to do this?

Maybe the app already exposes some API interface that can be used?

**Comment by devraj** [comment]
Without further context on this, it seems like a counter intuitive approach. That said you could use [background tasks](https://fastapi.tiangolo.com/tutorial/background-tasks/) and something like [httpx](https://github.com/encode/httpx). The big issue I see is timeouts so if the socket to the endpoint closes the handler would die.

The more I think about it, it would seem queue based workers would be a safer approach.
